crates/cli/src/main.rs

```diff
@@ -1,37 +1,45 @@
 //! cryo_cli is a cli for cryo_freeze
 
 use clap_cryo::Parser;
+use eyre::{Report, Result};
 
 mod args;
 mod parse;
 mod remember;
 mod run;
 
 pub use args::Args;
-use eyre::Result;
 
+/// Main entry point for the cryo_cli application.
+/// Initializes the application, parses arguments, and executes the run logic.
+/// Error handling is consistent across different build configurations.
 #[tokio::main]
-#[allow(unreachable_code)]
-#[allow(clippy::needless_return)]
 async fn main() -> Result<()> {
     let args = Args::parse();
     match run::run(args).await {
         Ok(Some(freeze_summary)) if freeze_summary.errored.is_empty() => Ok(()),
-        Ok(Some(_freeze_summary)) => std::process::exit(1),
+        Ok(Some(_)) => handle_error(),
+        Err(e) => handle_error_with_report(e),
         Ok(None) => Ok(()),
-        Err(e) => {
-            // handle debug build
-            #[cfg(debug_assertions)]
-            {
-                return Err(eyre::Report::from(e))
-            }
-
-            // handle release build
-            #[cfg(not(debug_assertions))]
-            {
-                println!("{}", e);
```

Comment:
this line should not be removed by the refactor

Reply:
added thanks for feedback